gateway/transport/interceptors/auth/auth.go

```diff
@@ -125,33 +131,109 @@ func (i *interceptor) StreamServerInterceptor(srv any, ss grpc.ServerStream, inf
 		}
 	// client proxy authentication (access token)
 	default:
-		sub, err := i.idp.VerifyAccessToken(bearerToken)
-		if err != nil {
-			log.Debugf("failed verifying access token, reason=%v", err)
-			return status.Errorf(codes.Unauthenticated, "invalid authentication")
-		}
-		userCtx, err := pguserauth.New().FetchUserContext(sub)
-		if userCtx.IsEmpty() {
+		apiKeyEnv := os.Getenv("API_KEY")
+		// this is a not so optimal solution, but due to the overall
+		// complexity of the authentication system, we decided to make this
+		// simple comparison on a optimistic way and if it fails, we fallback
+		// to the regular authentication flow with the IDP (see else stetament)
+		if apiKeyEnv != "" && apiKeyEnv == bearerToken {
+			log.Debug("Authenticating with API key")
+			orgID := strings.Split(bearerToken, "|")[0]
+			org, err := pgorgs.New().FetchOrgByID(orgID)
+			if err != nil || org == nil {
+				return status.Errorf(codes.Unauthenticated, "invalid authentication")
+			}
+			ctx := &pguserauth.Context{
+				OrgID:       orgID,
+				OrgName:     org.Name,
+				OrgLicense:  org.License,
+				UserUUID:    "API_KEY",
+				UserSubject: "API_KEY",
+				UserName:    "API_KEY",
+				UserEmail:   "API_KEY",
+				UserStatus:  "active",
+				UserGroups:  []string{"admin"},
+			}
+
+			gwctx := &GatewayContext{
+				UserContext: *ctx.ToAPIContext(),
+				BearerToken: bearerToken,
+			}
+
+			gwctx.UserContext.ApiURL = os.Getenv("API_URL")
+			connectionName := commongrpc.MetaGet(md, "connection-name")
+			conn, err := i.getConnection(connectionName, ctx)
 			if err != nil {
-				log.Error(err)
+				return err
 			}
-			return status.Errorf(codes.Unauthenticated, "invalid authentication")
-		}
-		gwctx := &GatewayContext{
-			UserContext: *userCtx.ToAPIContext(),
-			BearerToken: bearerToken,
-		}
-		gwctx.UserContext.ApiURL = i.idp.ApiURL
-		connectionName := commongrpc.MetaGet(md, "connection-name")
-		conn, err := i.getConnection(connectionName, userCtx)
-		if err != nil {
-			return err
-		}
-		if conn == nil {
-			return status.Errorf(codes.NotFound, "connection not found")
+			if conn == nil {
+				return status.Errorf(codes.NotFound, "connection not found")
+			}
+			gwctx.Connection = *conn
+			ctxVal = gwctx
+		} else {
```

Comment:
Remove this else block by adding a return in the if block above. It's unnecessary and hard to read it.